On Cygwin, /usr/bin == /bin && /usr/lib == /lib.  Prefer /usr.

Regarding the second hunk, AFAIK link libraries (symlinks on ELF platforms)
don't belong in /lib per the FHS, even if the runtime .so goes there.

--- origsrc/cmake-2.8.0/Modules/Platform/UnixPaths.cmake	2009-11-13 12:32:53.000000000 -0600
+++ src/cmake-2.8.0/Modules/Platform/UnixPaths.cmake	2009-11-24 22:14:00.604254500 -0600
@@ -33,7 +33,7 @@ GET_FILENAME_COMPONENT(_CMAKE_INSTALL_DI
 # search types.
 LIST(APPEND CMAKE_SYSTEM_PREFIX_PATH
   # Standard
-  /usr/local / /usr
+  /usr/local /usr /
 
   # CMake install location
   "${_CMAKE_INSTALL_DIR}"
@@ -74,7 +74,7 @@ LIST(APPEND CMAKE_SYSTEM_PROGRAM_PATH
   )
 
 LIST(APPEND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES
-  /lib /usr/lib /usr/lib32 /usr/lib64
+  /usr/lib /usr/lib32 /usr/lib64
   )
 
 LIST(APPEND CMAKE_C_IMPLICIT_INCLUDE_DIRECTORIES
